server/ladders-remote.ts

```diff
@@ -26,7 +26,7 @@ export class LadderStore {
 	 * client.
 	 */
 	async getTop(prefix?: string): Promise<[string, string] | null> {
-		return new Promise(resolve => resolve(null));
+		return Promise.resolve(null);
```

Comment:
@Zarel, Removing `async` make `typescript-eslint` complain because of the `require-await` since it's an async function with no Promise or `await`. In our case this doesn't make much sense because it needs to conform to the `LadderStore` interface.

Nevertheless, I think this is a reasonable rule as it's usually indicative of a unintended outcome refactor, so turning it off completely might not be the best idea.

Of course, you can just turn of the rule for the next line here and explain why. If you prefer it that way I'll change it, I don't really care.

Reply:
I would 100% suppress it, either with an eslint-ignore directive or by removing the require-await rule.

Reply:
I actually think the require-await rule is mostly pointless and we should just remove it.

Reply:
> Of course, you can just turn off the rule for the next line here and explain why. If you prefer it that way I'll change it, I don't really care.

I basically _always_ prefer turning off the rule, to writing worse code because the rule forces you to. This has come up many times in the past, and my position is always the same: add an eslint-ignore directive.

Reply:
> I actually think the require-await rule is mostly pointless and we should just remove it.

> I think this is a reasonable rule as it's usually indicative of a unintended outcome refactor, so turning it off completely might not be the best idea.

Reply:
I feel like it's usually indicative of implementing an interface that requires it, but fine, we can leave it if you want. Just do the ignore.